frontend/src/pages/Matching.vue

```diff
@@ -412,16 +392,32 @@ export default {
       "getAllTeachers",
       "matchStudentTeacherPairs",
     ]),
-    addMatchedPair() {
+    toggleSuggestedTeachersModal() {
+      Vue.nextTick().then(() => {
```

Comment:
This might raise question marks. The Modal toggling was happening before the `selected.sync` was running, meaning when the Modal opened, the `selectedStudent` data was not updated yet, leading either to empty student name or the student name that you **previously** clicked.

Putting the Modal opening after a tick resolved this, as we don't think there is any way to make the `selected.sync` run before the `@click`.